hydra-node/src/Hydra/HeadLogic/Outcome.hs

```diff
@@ -25,18 +27,20 @@ import Hydra.Tx (
   UTxOType,
   mkHeadParameters,
  )
+import Hydra.Tx.ContestationPeriod (ContestationPeriod)
 import Hydra.Tx.Crypto (MultiSignature, Signature)
 import Hydra.Tx.Environment (Environment (..))
 import Hydra.Tx.IsTx (ArbitraryIsTx)
+import Hydra.Tx.OnChainId (OnChainId)
 import Test.QuickCheck (oneof)
 import Test.QuickCheck.Arbitrary.ADT (ToADTArbitrary)
 
 -- | Analogous to inputs, the pure head logic "core" can have effects emited to
 -- the "shell" layers and we distinguish the same: effects onto the client, the
 -- network and the chain.
 data Effect tx
-  = -- | Effect to be handled by the "Hydra.API", results in sending this 'ServerOutput'.
-    ClientEffect {serverOutput :: ServerOutput tx}
+  = -- | Effect to be handled by the "Hydra.API", results in sending this 'StateChanged'.
+    ClientEffect {serverOutput :: StateChanged tx}
```

Comment:
The type and attribute names feel misleading to me.

Reply:
Would you call the field `stateChanged` ?

Reply:
@ffakenz is right. It's not only misleading, but IMO we should not even have this `ClientEffect` anymore.

Reply:
or eventToPersist